src/steamship/agents/mixins/transports/slack.py

```diff
@@ -120,8 +133,16 @@ def to_blocks(self) -> Optional[List[Block]]:
         ret = []
         for slack_block in self.blocks or []:
             for block in slack_block.to_blocks() or []:
+                # TODO (george): Having some sort of wrapper for Slack-specific Blocks might be nice
+                #   Or I guess the metadata block?
                 if self.channel:
                     block.set_chat_id(str(self.channel))
+                if self.ts:
+                    block.set_message_id(str(self.ts))
+                if self.user:
+                    block._one_time_set_tag(tag_kind=DocTag.CHAT, tag_name="userid", string_value=self.user)
```

Comment:
Ditto here w/r/t creating a `USER_ID` on `ChatTag`.  We might even include the transport (`slack`) for completeness while we're at it, too, so that reconstructing the data will make it clear where this message came from at the top level

a la:

```python
class ChatTag(str, Enum):
    # ship

    USER_ID = "user-id"
    USER_TRANSPORT = "user-transport"

```